spring-boot-actuator/src/main/java/org/springframework/boot/actuate/health/Neo4jHealthIndicator.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2012-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.actuate.health;
+
+import java.util.Collections;
+import java.util.Map;
+
+import org.neo4j.ogm.model.Result;
+import org.neo4j.ogm.session.Session;
+import org.neo4j.ogm.session.SessionFactory;
+
+/**
+ * {@link HealthIndicator} that tests the status of a Neo4j by executing a Cypher
+ * statement.
+ *
+ * @author Eric Spiegelberg
+ */
+public class Neo4jHealthIndicator extends AbstractHealthIndicator {
+
+	/** The key used to store Cypher execution results within the Health.Builder. */
+	public static final String NEO4J = "neo4j";
+
+	private SessionFactory sessionFactory;
+
+	/**
+	 * The default Cypher statement.
+	 */
+	private String cypher = "match (n) return count(n) as nodeCount";
+
+	/**
+	 * Create a new {@link Neo4jHealthIndicator} using the specified
+	 * {@link SessionFactory}.
+	 * @param sessionFactory the SessionFactory
+	 */
+	public Neo4jHealthIndicator(SessionFactory sessionFactory) {
+		this.sessionFactory = sessionFactory;
+	}
+
+	@Override
+	protected void doHealthCheck(Health.Builder builder) throws Exception {
+		Session session = this.sessionFactory.openSession();
+
+		Result result = session.query(this.cypher, Collections.emptyMap());
+		Iterable<Map<String, Object>> results = result.queryResults();
+
+		builder.up().withDetail(NEO4J, results);
```

Comment:
the entry will be something like

```
{
"neo4j" : 
  {
    "neo4j" : 8
  }
}
```

IMHO, the name should be `nodes` for this proprosal.